pkg/adapters/rules/kubernetes/kubernetes.go

```diff
@@ -214,51 +214,34 @@ func (a *Adapter) ListRules(f *a8api.RuleFilter) (*a8api.RulesSet, error) {
 	rules = f.Apply(rules)
 	rulesSet := &a8api.RulesSet{
 		Rules:    rules,
-		Revision: 1, // TODO: how to fill revision ?
-
+		Revision: int64(a.revision),
 	}
-	logger.Info("List rules: %v", rulesSet.Rules )
+
 	return rulesSet, nil
 }
 
 // addRule is the callback invoked by the Kubernetes cache when a rule API resource is added.
 func (a *Adapter) addRule(obj interface{}) {
-	rule, ok := obj.(*RoutingRule)
-	if !ok {
-		logger.Warnf("Invalid rule added: object is of type %T", obj)
-		return
-	}
-	if rule.Status.State == RuleStateInvalid {
-		logger.Warnf("Rule state is invalid. Rule %#v will not be added", rule)
-		return
-	} else if rule.Status.State == RuleStateValid {
-		a.revision++
-		a8Rule := &rule.Spec
-		ruleName := rule.Metadata.Name
-		a.rules[ruleName] = a8Rule
-		return
-	} else {
-		logger.Warnf("Rule state %s is undefiend . Rule %#v will not be added", rule.Status.State, rule)
-		return
-	}
+	a.storeRules(&obj)
 }
 
 // updateRule is the callback invoked by the Kubernetes cache when a rule API resource is updated.
 func (a *Adapter) updateRules(oldObj, newObj interface{}) {
-	oldRule, ok := oldObj.(*RoutingRule)
-	if !ok {
-		logger.Warnf("Invalid rule update: old object is of type %T\n", oldObj)
-		return
-	}
-	newRule, ok := newObj.(*RoutingRule)
+	a.storeRules(&newObj)
+
+}
+
+// storeRules is a helper function called by updateRules() and addRule() callback functions. is stores or updates rules
+// int the rules map of the adapter.
+func (a *Adapter) storeRules(obj *interface{}) {
```

Comment:
`*interface{}` type is confusing and unnecessary. Please declare the function argument to be `obj interface{}`. and invoke it with `a.storeRules(obj)` and _not_ `a.storeRules(&obj)`.